c++/src/kj/async-inl.h

```diff
@@ -1093,13 +1093,17 @@ inline PromiseForResult<Func, void> evalNow(Func&& func) {
 
 template <typename Func>
 inline PromiseForResult<Func, WaitScope&> startFiber(size_t stackSize, Func&& func) {
-  typedef _::FixVoid<_::ReturnType<Func, WaitScope&>> ResultT;
-
-  Own<_::FiberBase> intermediate = kj::heap<_::Fiber<Func>>(stackSize, kj::fwd<Func>(func));
-  intermediate->start();
-  auto result = _::PromiseNode::to<_::ChainPromises<_::ReturnType<Func, WaitScope&>>>(
-      _::maybeChain(kj::mv(intermediate), implicitCast<ResultT*>(nullptr)));
-  return _::maybeReduce(kj::mv(result), false);
+  #if KJ_NO_EXCEPTIONS
```

Comment:
Maybe this check could be in `FiberBase::FiberBase`, along-side the `__BIONIC__` check there that also calls `KJ_UNIMPLEMENTED`?

You'll also need to `#ifdef` out any code that actually uses `try`, `catch`, or `throw` when `KJ_NO_EXCEPTIONS` is on.

Reply:
I `#ifdef`ed out the fiber handling code in `waitImpl` which should cover all of it.

Reply:
You added a `throw` statement to `fiberCanceledException()`. That'll produce a compilation failure if exceptions are disabled with `-fno-exceptions`.

And per my other comment I think you need to add a `try`/`catch` in the fiber start function, since we don't want fibers to be required to catch `CanceledException` themselves... so that'll also need to be #ifdef'd out if exceptions are disabled.

Reply:
The start function being `armDepthFirst`? Also there is no `throw` statement in `fiberCanceledException`.

Reply:
Sorry, I had missed your latest changes when I wrote my last comment. It looks like you've got it mostly right. But `FiberBase::run()` looks like it won't compile if `-fno-exceptions` is enabled, so it needs to be #ifdef'd out. (That's what I meant by "start function".)

Reply:
Ah ok. :+1: